.buildkite/gem-push.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
```

Comment:
Nitpick

```suggestion
#!/bin/base -eu
```

or a comment on why we're not using the standard syntax (we should put on a post somewhere, maybe mobile.blog ? on why we use that)

Reply:
@mokagio I assume you meant `/bin/bash`?

Also for the `e` flag, I understand that it makes sense in most cases, but there are cases where it's OK & necessary to let the script handle the errors and decide whether to continue or not. I am OK with us defaulting to `eu` flags as a team, but if this was a personal project, I'd only use them when necessary and go with the defaults. 🤷

Reply:
> @mokagio I assume you meant /bin/bash?

Yep. Thank you. I updated the snipped.

Regarding the rest of your comment. My understanding is the same. There are some scripts in which we want to handle errors manually, therefore `#!/bin/bash -e` or `set -e` would not be appropriate. The same goes for `-u` for unassigned variables: we might want to prompt the user about them instead of failing.

I think my suggestion is appropriate in the context of this script, but I'll let @AliSoftware weight in.

Also, maybe my representation of what's our desired standard was incorrect. Should we simply go with "use `bash` instead of `sh`?"

Reply:
For scripts, my preference between `sh` & `bash` is always on the `bash` side, unless there is a reason for going with `sh`. (such as bash not being available)

Reply:
Fixed via https://github.com/wordpress-mobile/release-toolkit/pull/299/commits/6a1631416397a952bd46d410207993193552f717 by applying your GitHub suggestion (`bash -eu`)

---

I agree that in some cases we might want to handle the errors ourselves and thus `-e` is not the always right choice indeed. But in this case I think it is, as I have no custom error handling in this script, so better for it to stop at the first command that fails, rather than eat the error silently.

---

As for `-u`, it's opinionated but I think this one almost always makes sense, because even with that option, if you do want to handle cases where a variable is not set, or provide default values when a var is not provided by the user, there are safer ways to handle those (I love [this cheatsheet](https://devhints.io/bash) for good tips).
So IMHO it's better to use `-u` – which helped me catch typos in env vars in the past and find bugs many times – and use `MYVAR=${1:-default}` if you want to provide default values, rather than not use `-u` and risk issues (we've had a few in our Build Script phases of iOS projects in the past due to that `-u` missing and typos slipping). But that's just my own opinion 🤷